The switch to new channel layout api causes configure failure with ffmpeg plugin

<!-- Please fill the following if you have found an issue, delete if otherwise. -->
---

1) `uname -a`:
```
Darwin Mac-Studio 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:49 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6020 arm64
```

2) `cmus --version`:
N/A, trying to build v2.11

3) `cmus --plugins`:
N/A, ffmpeg plugin enabled

Reverting 3a7d9ee fixes this.
Trying to link ffmpeg 6.1.1. With edited configure script (removed the redirection to /dev/null), the failure message is:
```
...
:info:configure checking for FFMPEG_LIBS (pkg-config)... -L/opt/local/libexec/ffmpeg6/lib -lavformat -lavcodec
:info:configure checking for FFMPEG_CFLAGS (pkg-config)... -I/opt/local/libexec/ffmpeg6/include
:info:configure checking for header <libavcodec/avcodec.h>... yes
:info:configure checking for successful build of ffmpeg.c... .tmp-26-prog.c:242:10: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure         if (cc->ch_layout.order == AV_CHANNEL_ORDER_UNSPEC)
:info:configure             ~~  ^
:info:configure .tmp-26-prog.c:242:29: error: use of undeclared identifier 'AV_CHANNEL_ORDER_UNSPEC'
:info:configure         if (cc->ch_layout.order == AV_CHANNEL_ORDER_UNSPEC)
:info:configure                                    ^
:info:configure .tmp-26-prog.c:243:3: error: call to undeclared function 'av_channel_layout_default'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
:info:configure                 av_channel_layout_default(&cc->ch_layout, cc->ch_layout.nb_channels);
:info:configure                 ^
:info:configure .tmp-26-prog.c:243:34: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure                 av_channel_layout_default(&cc->ch_layout, cc->ch_layout.nb_channels);
:info:configure                                            ~~  ^
:info:configure .tmp-26-prog.c:243:49: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure                 av_channel_layout_default(&cc->ch_layout, cc->ch_layout.nb_channels);
:info:configure                                                           ~~  ^
:info:configure .tmp-26-prog.c:244:2: error: call to undeclared function 'av_opt_set_chlayout'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
:info:configure         av_opt_set_chlayout(swr, "in_chlayout",   &cc->ch_layout, 0);
:info:configure         ^
:info:configure .tmp-26-prog.c:244:2: note: did you mean 'av_opt_set_channel_layout'?
:info:configure /opt/local/include/libavutil/opt.h:710:5: note: 'av_opt_set_channel_layout' declared here
:info:configure int av_opt_set_channel_layout(void *obj, const char *name, int64_t ch_layout, int search_flags);
:info:configure     ^
:info:configure .tmp-26-prog.c:244:49: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure         av_opt_set_chlayout(swr, "in_chlayout",   &cc->ch_layout, 0);
:info:configure                                                    ~~  ^
:info:configure .tmp-26-prog.c:245:49: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure         av_opt_set_chlayout(swr, "out_chlayout",  &cc->ch_layout, 0);
:info:configure                                                    ~~  ^
:info:configure .tmp-26-prog.c:257:59: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure         ip_data->sf = sf_rate(cc->sample_rate) | sf_channels(cc->ch_layout.nb_channels);
:info:configure                                                              ~~  ^
:info:configure ./sf.h:52:29: note: expanded from macro 'sf_channels'
:info:configure #define sf_channels(val)        (((val) << SF_CHANNELS_SHIFT ) & SF_CHANNELS_MASK)
:info:configure                                    ^~~
:info:configure .tmp-26-prog.c:279:30: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure         channel_map_init_waveex(cc->ch_layout.nb_channels, cc->ch_layout.u.mask, ip_data->channel_map);
:info:configure                                 ~~  ^
:info:configure .tmp-26-prog.c:279:57: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure         channel_map_init_waveex(cc->ch_layout.nb_channels, cc->ch_layout.u.mask, ip_data->channel_map);
:info:configure                                                            ~~  ^
:info:configure .tmp-26-prog.c:395:40: error: no member named 'ch_layout' in 'struct AVCodecContext'
:info:configure                         output->buffer_used_len = res * cc->ch_layout.nb_channels * sizeof(int16_t);
:info:configure                                                         ~~  ^
:info:configure 12 errors generated.
:info:configure no
:info:configure configure failed.
```

--- a/ip/ffmpeg.c
+++ b/ip/ffmpeg.c
@@ -238,7 +238,7 @@ static int ffmpeg_open(struct input_plugin_data *ip_data)
 
 	/* Prepare for resampling. */
 	swr = swr_alloc();
-#if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR >= 60
 	if (cc->ch_layout.order == AV_CHANNEL_ORDER_UNSPEC)
 		av_channel_layout_default(&cc->ch_layout, cc->ch_layout.nb_channels);
 	av_opt_set_chlayout(swr, "in_chlayout",   &cc->ch_layout, 0);
@@ -253,7 +253,7 @@ static int ffmpeg_open(struct input_plugin_data *ip_data)
 	priv->swr = swr;
 
 	ip_data->private = priv;
-#if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR >= 60
 	ip_data->sf = sf_rate(cc->sample_rate) | sf_channels(cc->ch_layout.nb_channels);
 #else
 	ip_data->sf = sf_rate(cc->sample_rate) | sf_channels(cc->channels);
@@ -275,7 +275,7 @@ static int ffmpeg_open(struct input_plugin_data *ip_data)
 	}
 	swr_init(swr);
 	ip_data->sf |= sf_host_endian();
-#if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR >= 60
 	channel_map_init_waveex(cc->ch_layout.nb_channels, cc->ch_layout.u.mask, ip_data->channel_map);
 #else
 	channel_map_init_waveex(cc->channels, cc->channel_layout, ip_data->channel_map);
@@ -391,7 +391,7 @@ static int ffmpeg_fill_buffer(AVFormatContext *ic, AVCodecContext *cc, struct ff
 			if (res < 0)
 				res = 0;
 			output->buffer_pos = output->buffer;
-#if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR >= 60
 			output->buffer_used_len = res * cc->ch_layout.nb_channels * sizeof(int16_t);
 #else
 			output->buffer_used_len = res * cc->channels * sizeof(int16_t);

